### PR TITLE
Create post succeed modal refetch posts even onCancel

### LINKED ISSUE #1085: Posts are not displayed after creation if the confirmation modal is closed
#### Steps to recreate:
1) go to Feed/Help Board or Profile page or Org page
2) create a post
3) close the modal instead of clicking the `view post`
4) the post is not in the list of posts
5) refresh the page
6) the post is now in the list

#### Expected behavior:
post should be visible when the user closes the modal

#### Current Behavior: 

FightPandemics (2)

### PR DESCRIPTION
**Pull request by OzkanAbdullahoglu**
### What does this pull request aim to achieve? 💯

Resolves #1085

_Please be concise and link any related issue(s):_

- [x] I have checked that no one else is working on similar changes.
- [x] I have read the latest [**README**](https://github.com/FightPandemics/FightPandemics/blob/master/README.md) and understand the development workflow.
- [x] The name of this branch is: **`feature/<branch_name>`** (need to have cloned the repo not forked).
- [x] This branch is rebased/merged with the **latest master**.
- [x] There are no merge conflicts.
- [x] There are no console warnings and errors.
- [x] On the right hand side of this PR, I have linked the appropriate issue(s) under "Issues" and added a project progress under "Projects".

### CODE REVIEW

File: client/src/pages/Feed.js [hunk 1]
```diff
@@ -575,6 +575,14 @@ const Feed = (props) => {
       }
     }
   };
+
+  const onCancelModal = (currentStep) => {
+    dispatchAction(TOGGLE_STATE, "showCreatePostModal");
+    if (currentStep == 4) {
+      loadPosts();
```
Comment (nr431207) [inline client/src/pages/Feed.js:582]: this works just fine, I think we can improve this by doing the `currentStep === 4` within the `CreatePost` component instead so we wont have to check it in all the other components.

File: client/src/pages/OrganisationProfile.js [hunk 2]
```diff
@@ -262,6 +262,13 @@ const OrganisationProfile = () => {
     }
   };
 
+  const onCancelModal = (currentStep) => {
+    setModal(false);
+    if (currentStep == 4) {
```
Comment (nr431207) [inline client/src/pages/OrganisationProfile.js:267]: same as above, we can do this in check in `createComponent`

File: client/src/pages/Profile.js [hunk 3]
```diff
@@ -238,6 +238,13 @@ const Profile = ({
     }
   };
 
+  const onCancelModal = (currentStep) => {
+    setModal(false);
+    if (currentStep == 4) {
```
Comment (nr431207) [inline client/src/pages/Profile.js:243]: same as above

**Comment by nr431207** [comment]
I tested the buildURL, in Feeds the first time I create a post I can see the post when I close the modal, but when I create a second time, I dont see it unless I refresh.. not sure why this is happening tho.. in orgProfile and userProfile it is working fine though.

**Comment by OzkanAbdullahoglu** [comment]
> I tested the buildURL, in Feeds the first time I create a post I can see the post when I close the modal, but when I create the second time, I dont see it unless I refresh.. not sure why this is happening tho.. in orgProfile and userProfile it is working fine though.

I think It might be related with below issue ? 
https://github.com/FightPandemics/FightPandemics/issues/875

**Comment by nr431207** [comment]
> > I tested the buildURL, in Feeds the first time I create a post I can see the post when I close the modal, but when I create the second time, I dont see it unless I refresh.. not sure why this is happening tho.. in orgProfile and userProfile it is working fine though.
> 
> I think It might be related with below issue ?
> #875

I checked on my local to make sure(using your changes), but it's not related to this issue, its just not refetched for some reason...

**Comment by OzkanAbdullahoglu** [comment]
> > > I tested the buildURL, in Feeds the first time I create a post I can see the post when I close the modal, but when I create the second time, I dont see it unless I refresh.. not sure why this is happening tho.. in orgProfile and userProfile it is working fine though.
> > 
> > 
> > I think It might be related with below issue ?
> > #875
> 
> I checked on my local to make sure(using your changes), but it's not related to this issue, its just not refetched for some reason...

I ll look into this too...

**Comment by OzkanAbdullahoglu** [comment]
> > > > I tested the buildURL, in Feeds the first time I create a post I can see the post when I close the modal, but when I create the second time, I dont see it unless I refresh.. not sure why this is happening tho.. in orgProfile and userProfile it is working fine though.
> > > 
> > > 
> > > I think It might be related with below issue ?
> > > #875
> > 
> > 
> > I checked on my local to make sure(using your changes), but it's not related to this issue, its just not refetched for some reason...
> 
> I ll look into this too...

I implemented proper refetch oncancel. But in feed.js  we do not have an exact control on what we fetch firstly from `loadedPosts` . Which come from `const baseURL = `/api/posts?limit=${limit}&skip=${skip}`;` .  Can we get some help on this ? @joshmorel

**Comment by joshmorel** [comment]
@OzkanAbdullahoglu (cc @nr431207 @idjevm ) So I updated the `Feed.js` code, so if the user closes the modal it will refetch the posts (and resets filters/pagination).  `loadPosts` listens to changes in other state so it is not the right thing here to call directly.

**Comment by nr431207** [comment]
> @OzkanAbdullahoglu (cc @nr431207 @idjevm ) So I updated the `Feed.js` code, so if the user closes the modal it will refetch the posts (and resets filters/pagination). `loadPosts` listens to changes in other state so it is not the right thing here to call directly.

@joshmorel aah it makes sense! thank you for these changes!

**Comment by joshmorel** [comment]
@idjevm no sorting after location (if logged in or location filter set) is currently indeterminate. This is ongoing or future discussion so out scope for this issue.

**Comment by idjevm** [comment]
@joshmorel does that apply for an user new posts as well or? Because if I have a few posts, It will be added but it will be a few pages down, I will have to dig on the feed to see the newly added post....

**Comment by joshmorel** [comment]
@idjevm yes except on the profile pages. I can send you a link to the last convo in slack about this. Regardless or preferred behaviour this is not the PR to address this.

**Comment by OzkanAbdullahoglu** [comment]
and maybe to implement a go to the top button would be nice and this can be discussed under the same title later on? @joshmorel @idjevm @nr431207

**Pull request merged**